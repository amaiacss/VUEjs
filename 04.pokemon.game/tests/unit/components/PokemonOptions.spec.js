import { shallowMount } from '@vue/test-utils'
// importamos el sujeto de pruebas
import PokemonOptions from '@/components/PokemonOptions'

//importar el array de un archivo mock 
import { mockPokemons } from '../mocks/pokemons.mock'



describe('PokemonOptions component', () => {
    
    let wrapper
    beforeEach(() => {
        wrapper = shallowMount( PokemonOptions, {
            props: {
                pokemons: mockPokemons // esto es el array para poder reutilizar en cualquier prueba

            }
        }) 
    })

    test('Debe hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
        
    })        
    
    test('Debe mostrar las 4 opciones correctamente', () => {
        // que existan 4 li y cada li debe tener su nombre del pokemon
        const liTags = wrapper.findAll( 'li' )
        // primero preguntamos si exsite el elemento, y si son 4
        expect( liTags.length ).toBe(4)
        // Evaluar cada uno de esos li y ver que tiene su respectivo valor
        expect( liTags[0].text() ).toBe('bulbasaur')
        expect( liTags[1].text() ).toBe('ivysaur')
        expect( liTags[2].text() ).toBe('venusaur')
        expect( liTags[3].text() ).toBe('charmander')           
    })       
    
    test('Debe emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => {
        const [ li1, li2, li3, li4  ] = wrapper.findAll('li')
        //hacer click sobre el elemento
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        console.log( wrapper.emitted('selection') )
        // Se espera que por lo menos se tiene que haber emitido una vez
        expect( wrapper.emitted('selection').length ).toBe(4)
        // emitido con el valor esperado
        expect( wrapper.emitted('selection')[0] ).toEqual([1])
        expect( wrapper.emitted('selection')[1] ).toEqual([2])
        expect( wrapper.emitted('selection')[2] ).toEqual([3])
        expect( wrapper.emitted('selection')[3] ).toEqual([4])

           
    })
    
})