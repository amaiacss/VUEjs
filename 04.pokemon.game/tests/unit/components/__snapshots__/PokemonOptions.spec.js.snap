// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>
      <!-- Puede recibir dos argumentos (el primero es obligatorio), el 2ยบ es opcinal -->bulbasaur
    </li>
    <li>
      <!-- Puede recibir dos argumentos (el primero es obligatorio), el 2ยบ es opcinal -->ivysaur
    </li>
    <li>
      <!-- Puede recibir dos argumentos (el primero es obligatorio), el 2ยบ es opcinal -->venusaur
    </li>
    <li>
      <!-- Puede recibir dos argumentos (el primero es obligatorio), el 2ยบ es opcinal -->charmander
    </li>
  </ul>
</div>
`;
